main/commons/src/main/java/org/cryptomator/common/mountpoint/CustomMountPointChooser.java

```diff
@@ -0,0 +1,80 @@
+package org.cryptomator.common.mountpoint;
+
+import org.cryptomator.common.vaults.MountPointRequirement;
+import org.cryptomator.common.vaults.Vault;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.DirectoryNotEmptyException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.NotDirectoryException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+public class CustomMountPointChooser implements MountPointChooser {
+
+	private static final Logger LOG = LoggerFactory.getLogger(CustomMountPointChooser.class);
+
+	private final Vault vault;
+
+	public CustomMountPointChooser(Vault vault) {
+		this.vault = vault;
+	}
+
+	@Override
+	public Optional<Path> chooseMountPoint() {
+		//VaultSettings#getCustomMountPath already checks whether the saved custom mountpoint should be used
+		return this.vault.getVaultSettings().getCustomMountPath().map(Paths::get);
+	}
+
+	@Override
+	public boolean prepare(Path mountPoint) throws InvalidMountPointException {
+		MountPointRequirement requirement = this.vault.getMountPointRequirement();
```

Comment:
Resolved by 8680bd1deecc4a1ebce712d06e7e8083e41e7445.

Reply:
> Therefore, i suggest to refactor moutpoint chooser to take the volume enum as a parameter.

Do you mean this specific MPC or all MPCs?

Reply:
In every chooser where they are used. Currently, i think, the custom one is the only one.

I may be wrong, but from my point of the view, the volume and the OS define the requirements, and therefore not the vault should return them, but the used volume. With dagger it can be injected.

Reply:
Resolved by d29508431242d4793eabc5ff09d7a0b690cc49c7, 22438d1ebae892d303bbb3719b4482f5fde71d55 and b5efe39eb88a99c8d3266d36dc63f8fd5036d265.